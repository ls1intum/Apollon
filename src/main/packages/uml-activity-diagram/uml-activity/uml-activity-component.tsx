import React, { FunctionComponent, SFC } from 'react';
import { Text } from '../../../components/controls/text/text';
import { UMLActivity } from './uml-activity';

export const UMLActivityComponent: FunctionComponent<Props> = ({ element, children, scale }) => (
  <g>
    <rect rx={10 * scale} ry={10 * scale} width="100%" height="100%" stroke={element.strokeColor || 'black'} />
    <Text y={20 * scale} fill={element.textColor}>
      {element.name}
    </Text>
    {children}
  </g>
);

interface Props {
  element: UMLActivity;
  scale: number;
}
