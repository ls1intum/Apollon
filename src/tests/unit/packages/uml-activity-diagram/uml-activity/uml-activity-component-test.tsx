import { UMLActivityComponent } from '../../../../../main/packages/uml-activity-diagram/uml-activity/uml-activity-component';
import { UMLActivity } from '../../../../../main/packages/uml-activity-diagram/uml-activity/uml-activity';
import * as React from 'react';
import { wrappedRender } from '../../../test-utils/render';

it('render the uml-activity-component', () => {
  const activity: UMLActivity = new UMLActivity({ name: 'TestActivityComponent' });
  const { getByText, baseElement } = wrappedRender(
    <svg>
      <UMLActivityComponent element={activity} />
    </svg>,
  );
  expect(getByText(activity.name)).toBeInTheDocument();
  expect(baseElement).toMatchSnapshot();
});
