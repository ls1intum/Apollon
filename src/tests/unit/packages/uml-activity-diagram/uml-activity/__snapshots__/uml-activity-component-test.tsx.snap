// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the uml-activity-component 1`] = `
<body>
  <div>
    <svg>
      <g>
        <rect
          class="sc-fqkvVR kprGcp"
          fill="white"
          height="100%"
          rx="10"
          ry="10"
          stroke="black"
          width="100%"
        />
        <text
          dominant-baseline="middle"
          font-weight="bold"
          pointer-events="none"
          text-anchor="middle"
          x="50%"
          y="20"
        >
          TestActivityComponent
        </text>
      </g>
    </svg>
  </div>
</body>
`;
